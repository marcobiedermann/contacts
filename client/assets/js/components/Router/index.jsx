import { ConnectedRouter } from 'connected-react-router';
import PropTypes from 'prop-types';
import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Layout from '../Layout';
import IndexPage from '../../pages/Index';
import ContactsPage from '../../pages/Contacts';
import NewPage from '../../pages/New';
import history from '../../store/history';

const Router = ({
  addContact,
  updateContact,
  removeContact,
}) => (
  <ConnectedRouter history={history}>
    <Layout>
      <Switch>
        <Route
          path="/contacts"
          component={() => (
            <ContactsPage
              updateContact={updateContact}
              removeContact={removeContact}
            />
          )}
        />
        <Route
          path="/new"
          component={() => (
            <NewPage addContact={addContact} />
          )}
        />
        <Route
          path="/"
          component={IndexPage}
        />
      </Switch>
    </Layout>
  </ConnectedRouter>
);

Router.propTypes = {
  addContact: PropTypes.func,
  updateContact: PropTypes.func,
  removeContact: PropTypes.func,
};

Router.defaultProps = {
  addContact: () => {},
  updateContact: () => {},
  removeContact: () => {},
};

export default Router;
