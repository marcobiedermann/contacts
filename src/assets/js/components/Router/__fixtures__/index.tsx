import React, { FC } from 'react';
import { RouterProps } from '..';

const HomePage: FC = () => <div>Home</div>;

const props: RouterProps = {
  routes: [
    {
      component: HomePage,
      name: 'Home',
      path: '/',
    },
  ],
};

export default props;
