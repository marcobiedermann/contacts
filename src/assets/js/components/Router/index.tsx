import { ConnectedRouter } from 'connected-react-router';
import React from 'react';
import { Route, Switch } from 'react-router-dom';
import * as routes from '../../constants/routes';
import ContactsPage from '../../containers/ContactsPage';
import LoginPage from '../../containers/LoginPage';
import LogoutPage from '../../containers/LogoutPage';
import ContactNewPage from '../../pages/ContactNewPage';
import IndexPage from '../../pages/IndexPage';
import RegisterPage from '../../pages/RegisterPage';
import history from '../../store/history';
import Layout from '../Layout';

const Router = () => (
  <ConnectedRouter history={history}>
    <Layout>
      <Switch>
        <Route path={routes.CONTACTS} component={ContactsPage} />
        <Route path={routes.LOGIN} component={LoginPage} />
        <Route path={routes.NEW} component={ContactNewPage} />
        <Route path={routes.REGISTER} component={RegisterPage} />
        <Route path={routes.LOGOUT} component={LogoutPage} />
        <Route path={routes.INDEX} component={IndexPage} />
      </Switch>
    </Layout>
  </ConnectedRouter>
);

export default Router;
