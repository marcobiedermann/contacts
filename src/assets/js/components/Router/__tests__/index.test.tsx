import { render } from '@testing-library/react';
import React from 'react';
import Router from '..';
import props from '../__fixtures__';

describe('router component', () => {
  it('should render correctly', () => {
    expect.assertions(1);

    const { container } = render(<Router {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
