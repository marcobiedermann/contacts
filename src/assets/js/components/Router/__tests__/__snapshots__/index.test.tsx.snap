// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`router component should render correctly 1`] = `
<div
  class="layout"
>
  <header
    class="header"
  >
    <div
      class="grid"
    >
      <nav
        class="navigation"
      >
        <ul
          class="navigation__list navigation__list--inline"
        >
          <li>
            <a
              href="/contacts"
            >
              pages:contacts
            </a>
          </li>
          <li>
            <a
              href="/contacts/new"
            >
              pages:new
            </a>
          </li>
          <li>
            <a
              href="/settings"
            >
              pages:settings
            </a>
          </li>
          <li>
            <a
              href="/login"
            >
              pages:login
            </a>
          </li>
          <li>
            <a
              href="/logout"
            >
              pages:logout
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
  <main
    class="main"
  >
    <div
      class="grid"
    >
      <div>
        Home
      </div>
    </div>
  </main>
  <footer>
    <div
      class="grid"
    >
      Footer
    </div>
  </footer>
</div>
`;
